htdocs/core/lib/functions.lib.php

```diff
@@ -13702,7 +13702,7 @@ function forgeSQLFromUniversalSearchCriteria($filter, &$errorstr = '', $noand =
 		$filter = '(' . $filter . ')';
 	}
 
-	$regexstring = '\(([a-zA-Z0-9_\.]+:[<>!=insotlke]+:[^\(\)]+)\)';	// Must be  (aaa:bbb:...) with aaa is a field name (with alias or not) and bbb is one of this operator '=', '<', '>', '<=', '>=', '!=', 'in', 'notin', 'like', 'notlike', 'is', 'isnot'
+	$regexstring = '\(([a-zA-Z0-9_\.\-]+:[<>!=insotlke]+:[^\(\)]+)\)';	// Must be  (aaa:bbb:...) with aaa is a field name (with alias or not) and bbb is one of this operator '=', '<', '>', '<=', '>=', '!=', 'in', 'notin', 'like', 'notlike', 'is', 'isnot'
```

Comment:
I think you introduce the "-" char to allow fields like t.datef-year
I think for filter, we should not have the date-year or date-month  fields, only the date. This variant were introduced for the X axis but is not interesting for filter because to get a good SQL syntax it is better to do a date > YYYY0101 and date < YYYY1231 instead of dateformat. So lets just ask date with YYYY MM DD

In a next next we will allow to enter year > ... but we must forge date >  YYYY0101